gossip/src/cluster_info.rs

```diff
@@ -1513,7 +1552,8 @@ impl ClusterInfo {
         let self_info = CrdsValue::new_signed(self_info, &self.keypair());
         let pulls = pulls
             .into_iter()
-            .flat_map(|(peer, filters)| repeat(peer.gossip).zip(filters))
+            .filter_map(|(peer, filters)| Some((peer.gossip().ok()?, filters)))
```

Comment:
this pattern (`filter_map(|v| Some(...?)`) catches me up every time. prolly prefer the transposition of `.ok().map(|v| ...)` to it

Reply:
This is an iterator, so `filter_map` is inevitable.
I think the alternative of putting lambda inside `filter_map` is less readable than the current one.